src/chainstate/stacks/db/blocks.rs

```diff
@@ -4245,6 +4252,26 @@ impl StacksChainState {
                 )
             };
 
+            // get previous burn block stats
+            let (prev_burn_block_hash, prev_burn_block_height, prev_burn_block_timestamp) =
+                match SortitionDB::get_block_snapshot_consensus(
+                    burn_dbconn,
+                    &parent_consensus_hash,
+                )? {
+                    Some(sn) => (
+                        sn.burn_header_hash,
+                        sn.block_height as u32,
+                        sn.burn_header_timestamp,
+                    ),
+                    None => {
+                        // shouldn't happen
+                        panic!(
+                            "CORRUPTION: staging block {}/{} does not correspond to a burn block",
+                            &parent_consensus_hash, &parent_block_hash
+                        );
```

Comment:
Does this panic on the first block? (i.e., if `parent_consensus_hash` is `FIRST_BURNCHAIN_CONSENSUS_HASH`)?

Reply:
I left a panic, but now I'm also explicitly checking whether or not the block is the first.